packages/react-native-bridge/android/react-native-bridge/src/main/java/org/wordpress/mobile/ReactNativeGutenbergBridge/RNReactNativeGutenbergBridgeModule.java

```diff
@@ -554,14 +555,44 @@ private ConnectionStatusCallback requestConnectionStatusCallback(final Callback
         };
     }
 
+    @ReactMethod
+    public void showAndroidSoftKeyboard() {
+        Activity currentActivity = mReactContext.getCurrentActivity();
+        if (currentActivity != null) {
+            // Define the delay in milliseconds to prevent trying to show the keyboard right when a UI change happened.
+            int delayMillis = 500;
```

Comment:
@geriux I noticed this delay when dismissing the bottom sheet and was wondering what would be the impact of reducing it. Seems using a value below 100 ms (when using an Android emulator), the keyboard won't be shown. Based on the inline comment, I understood that the delay was used to avoid conflicts with UI but not that it's a requirement to ensure the keyboard is displayed. What do you think about clarifying this in the comment?

On the other hand, I'm concerned that in low-end devices this value (i.e. 500ms) would be too low. I wonder if we could figure out a way to detect when the keyboard can be shown 🤔.

Reply:
> Thanks for the feedback! I tried different approaches but it's complicated since we can easily detect when the Dialog view is fully hidden from the native side.

Yeah, I imagined it wouldn't be easy. Thanks for exploring alternatives 🙇 !

> Taken from [this code](https://github.com/wordpress-mobile/WordPress-Android/blob/6e653b566c96970ebbad1f88e4574ac6a5c518fb/WordPress/src/main/java/org/wordpress/android/util/extensions/ViewExtensions.kt#L46-L79). Which worked pretty well but I had to add a delay for the main inserter menu, I think it's because is shown in Full screen and how it is implemented that from the moment it gets unmounted, the keyboard wouldn't show unless we delay it a bit.

That's interesting, I didn't know there was a similar logic in WP-Android. Regarding full-screen modals, sounds good to me to leave it as-is. In the worst case, in case the delay is not long enough for the device, we won't be showing the keyboard after dismissing the inserter modal which isn't a critical issue.

> For other and more used cases while editing (Block settings, Alignments, link settings) the keyboard is shown pretty quickly without any fixed delays.

Nice, it's great to see that there's a way to wait for the exact moment to show the keyboard 🏅.

> Let me know what you think about this approach, thanks!

I'll add some comments in the PR review to get a better understand of the logic, but the current approach looks good to me.